crd-generator/api-v2/src/test/java/io/fabric8/crdv2/generator/v1/JsonSchemaDefaultValueTest.java

```diff
@@ -0,0 +1,175 @@
+  }
+
+  @Test
+  @DisplayName("Default annotation for String")
+  void stringFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForString._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .extracting(JsonNode::asText)
+        .isEqualTo("string-from-default-annotation");
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for boxed Boolean")
+  void booleanBoxedFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForBoxedBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("Default annotation for boxed Boolean")
+  void booleanBoxedFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForBoxedBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for boolean")
+  void booleanFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("Default annotation for boolean")
+  void booleanFromDefaultAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultAnnotationForBoolean._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(BooleanNode.class)
+        .extracting(JsonNode::asBoolean)
+        .isEqualTo(true);
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for int")
+  void intFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForInt._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(LongNode.class)
+        .extracting(JsonNode::asInt)
+        .isEqualTo(1337);
+  }
+
+  @Test
+  @DisplayName("JsonProperty defaultValue annotation for long")
+  void longFromJsonPropertyAnnotation() {
+    assertThat(JsonSchema.from(ClassInTest.class).getProperties())
+        .extracting("defaultValueForLong._default")
+        .asInstanceOf(InstanceOfAssertFactories.type(JsonNode.class))
+        .isInstanceOf(LongNode.class)
+        .extracting(JsonNode::asLong)
+        .isEqualTo(1337L);
```

Comment:
Would it be possible to also generate the resulting YAML file?

I'm interested in checking the `L` and `d` suffixes

Reply:
The suffixes here I added them to highlight that these values are equal to the expected type, to make it clear that a long is a long and a double is not a double.
How it renders in the CRD is not about how the value is stored in a JsonNode, but how the serializer deals with those. Basically they'll be rendered as any other number in a JSON->YAML document.

There's also an approval test modified that you can check:
https://github.com/fabric8io/kubernetes-client/blob/bee2f0154aa088a0343e2a045766c6ed91701e28/crd-generator/test/src/test/resources/io/fabric8/crd/generator/approvaltests/CRDGeneratorApprovalTest.approvalTest.annotateds.samples.fabric8.io.v1.approved.yml#L25-L36

I noticed that I didn't add a specific test for doubles, but the expected value should be an unquoted number with a `.` in case it has decimals.

Reply:
https://github.com/fabric8io/kubernetes-client/pull/6670